src/message_reader.cc

```diff
@@ -77,16 +77,16 @@ uint32_t DelimiterToSize(const unsigned char* delimiter) {
 
 }  // namespace
 
-std::unique_ptr<pbio::ZeroCopyInputStream> MessageReader::NextMessage() {
+std::unique_ptr<pbio::ZeroCopyInputStream> MessageReader::NextMessage(unsigned char* delimiter) {
   if (Finished()) {
     // The stream has ended
     return nullptr;
   }
 
   // Check if we have the current message size. If not try to read it.
   if (!have_current_message_size_) {
-    const size_t kDelimiterSize = 5;
-    if (in_->BytesAvailable() < static_cast<pb::int64>(kDelimiterSize)) {
+    if (in_->BytesAvailable()
```

Comment:
If `have_current_message_size_` is tgrue,  line 102, ReadSteram into delimitor.  is not called.

if you want to preserve 5 bytes delimitor, you have to  save it in the class member.